mlir/utils/jenkins/Jenkinsfile.miopen-test

```diff
@@ -1,65 +1,101 @@
+                                cmakeBuild buildDir: 'build',\
+                                   buildType: 'Release',\
+                                   installation: 'InDocker',\
+                                   cmakeArgs: '''-DMIOPEN_USE_MLIR=On
+-DCMAKE_CXX_COMPILER=/opt/rocm/llvm/bin/clang++
+-DCMAKE_C_COMPILER=/opt/rocm/llvm/bin/clang
+-DBUILD_DEV=On
+-DMIOPEN_TEST_FLAGS=\'--verbose --disable-verification-cache\'
+'''
+                                sh 'cd build; make -j $(nproc) MIOpenDriver'
+                            }
+                        }
+                    }
+                    stage("Test MIOpen config") {
+                        steps {
+                            echo "Config is ($DTYPE, $LAYOUT, $XDLOPS, dir: $DIRECTION)"
+                            dir('MIOpen/build/') {
+                                sh """
+bash ${WORKSPACE}/mlir/utils/jenkins/miopen-tests/miopen_validate.sh \
+--layout $LAYOUT \
+--direction $DIRECTION \
+--dtype $DTYPE \
+${XDLOPS == '1' ? '--xdlops' : '--no-xdlops'} <${WORKSPACE}/mlir/utils/jenkins/miopen-tests/resnet50-miopen-configs"""
+                            }
+                        }
+                    }
+                }
             }
         }
-        stage('Build MIOpenDriver and Test') {
-            steps {
-                checkout scm
-                sh '''
-                    cd ~
-                    git clone https://github.com/ROCmSoftwarePlatform/MIOpen.git
-                    cd MIOpen
-                    git checkout igemm-bin-xdlops
-                    mkdir -p build && cd build
-                    CXX=/opt/rocm/llvm/bin/clang++ cmake -DMIOPEN_USE_MLIR=On \
-                         -DBUILD_DEV=On \
-                         -DCMAKE_BUILD_TYPE=release \
-                         -DMIOPEN_TEST_FLAGS='--verbose \
-                         --disable-verification-cache' ..
-                    make -j MIOpenDriver
-                    bash $WORKSPACE/mlir/utils/jenkins/miopen-tests/resnet50_fp16_nchw.sh 
-                    bash $WORKSPACE/mlir/utils/jenkins/miopen-tests/resnet50_fp32_nchw.sh 
-                    bash $WORKSPACE/mlir/utils/jenkins/miopen-tests/resnet50_fp16_nhwc.sh 
-                    bash $WORKSPACE/mlir/utils/jenkins/miopen-tests/resnet50_fp32_nhwc.sh 
-                    bash $WORKSPACE/mlir/utils/jenkins/miopen-tests/resnet50_fp16_nchw_xdlop.sh 
-                    bash $WORKSPACE/mlir/utils/jenkins/miopen-tests/resnet50_fp32_nchw_xdlop.sh 
-                    bash $WORKSPACE/mlir/utils/jenkins/miopen-tests/resnet50_fp16_nhwc_xdlop.sh 
-                    bash $WORKSPACE/mlir/utils/jenkins/miopen-tests/resnet50_fp32_nhwc_xdlop.sh 
-                    cd ../../
-                    rm -rf MIOpen
-                '''
-            }
-        }
-    }
-    post {
-        always {
-            cleanWs()
```

Comment:
How does it cleanup workspace now?

> We can consider enabling cleaning before builds

Why clean before not after builds?

Reply:
The workspace is not cleaned up now - I'm not yet sure how to execute cleanup on all nodes. If I find a way to do that, I'll update.

Reply:
@krzysz00 Ping @okakarpa if you are not sure. He is the domain expert on this.

Reply:
Some googling shows that https://plugins.jenkins.io/hudson-wsclean-plugin/ exists - maybe we want to use it?

Reply:
@okakarpa Could you comment on that plugin @krzysz00 mentioned about in clean up workspace from distributed environment?